miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java

```diff
@@ -878,9 +878,13 @@ protected SampleDto asDto(Sample model) {
 
     @Override
     protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) {
-      config.putPOJO("targetSampleClass", Dtos.asDto(sampleClass));
-      config.putPOJO("sourceSampleClass", Dtos.asDto(sampleClass));
-      config.put("dnaseTreatable", sampleClass.getDNAseTreatable());
+      if (sampleClass != null) {
+        config.putPOJO("targetSampleClass", Dtos.asDto(sampleClass));
+        config.putPOJO("sourceSampleClass", Dtos.asDto(sampleClass));
+        config.put("dnaseTreatable", sampleClass.getDNAseTreatable());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "dnaseTreatable" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

Reply:
Would be ideal for all valid config keys, but maybe not critical